pkg/testutils/unstructured.go

```diff
@@ -0,0 +1,116 @@
+package testutils
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/pkg/util"
+)
+
+func DeleteCR(ctx context.Context, clnt client.Client, obj client.Object) error {
+	if err := clnt.Delete(ctx, obj); util.IsNotFound(err) {
+		return nil
+	}
+	if err := clnt.Get(ctx, client.ObjectKey{Name: obj.GetName(), Namespace: obj.GetNamespace()}, obj); err != nil {
+		if util.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	return fmt.Errorf("%s/%s: %w", obj.GetNamespace(), obj.GetName(), ErrNotDeleted)
+}
+
+func DeleteCRWithGVK(ctx context.Context, clnt client.Client, name, namespace, group, version, kind string) error {
+	obj, err := GetCR(ctx, clnt, client.ObjectKey{Name: name, Namespace: namespace}, schema.GroupVersionKind{
+		Group:   group,
+		Version: version,
+		Kind:    kind,
+	})
+	if err != nil {
+		return err
+	}
+	return DeleteCR(ctx, clnt, obj)
+}
+
+func GetCR(ctx context.Context, clnt client.Client,
+	objectKey client.ObjectKey,
+	gvk schema.GroupVersionKind,
+) (*unstructured.Unstructured, error) {
+	obj := &unstructured.Unstructured{}
+	obj.SetGroupVersionKind(gvk)
+	if err := clnt.Get(ctx, objectKey, obj); err != nil {
+		if util.IsNotFound(err) {
+			return obj, nil
+		}
+		return nil, err
+	}
+	return obj, nil
+}
+
+func IsResourceVersionSame(ctx context.Context, clnt client.Client,
```

Comment:
for a function with boolean as return value, use `is` or `if` as prefix is a common naming pattern, you can search more detail https://www.bing.com/search?q=+boolean+function+naming+pattern&qs=n&form=QBRE&sp=-1&lq=0&pq=+boolean+function+naming+pattern&sc=11-32&sk=&cvid=425A6121A15A43898F423A6AD303CD29&ghsh=0&ghacc=0&ghpl=

Reply:
Yes but this one is used for the tests. so we have this naming convention here because it would be used with Eventually or Consistently then it would be read as a statement. It is similar to CRIsInState for example.

Reply:
something defined as **naming conversion** should be consider as a widely accepted concept. I don't think we should treat the naming of the function in test as a special case, so I can't say this `CRIsInState` is so called naming conversion.

But of course, we can have a team agreement that we have some teamwised consensus for certain naming rules, but then lets bring this to the team before get decision.